mlflow/protos/service.proto

```diff
@@ -573,7 +573,7 @@ message RestoreExperiment {
   option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
 
   // Identifier to get an experiment
-  optional int64 experiment_id = 1 [(validate_required) = true];
+  optional string experiment_id = 1 [(validate_required) = true];
```

Comment:
This will be a breaking change in the contracts. Generally changing type in a proto file means renaming the old field and introducing the a new indexed field with the correct name and type.
e.g.
optional int64 deprecated_experiment_id = 1 [(validate_required) = true];
optional string experiment_id = 2 [(validate_required) = true];

Reply:
Although it is acceptable to make a breaking change prior to MLflow 1.0, we should double-check the behavior with respect to serialized proto JSON. I suspect that int-y strings will be parsed correctly by the Python deserializer.

Could we run such a test -- running this code against an old version of the server, and confirming that it works?

Reply:
I have not aded  old server test. But I did add reading string experiment_id as int. Specifically for Metric timestamp(any int64) since the same deserializer is used throughout. 

If that is not enough, I can keep an old service.proto and it's generated service_pb2.py, and parametrize handlers.py so we can test against both versions in test_handlers.py and test_rest_tracking.py

Reply:
I ran the new client against an old server with the updated backcompat helper and it passed the tests.